pkg/graph/resolvers/system_intake_actions.go

```diff
@@ -628,3 +628,200 @@ func CreateSystemIntakeActionNotITGovRequest(
+
+	// save action (including additional info for email, if any)
+	errGroup := new(errgroup.Group)
+	action := lcidactions.GetUpdateLCIDAction(*intake, input.ExpiresAt, input.NextSteps, input.Scope, input.CostBaseline, *adminUserInfo)
+	errGroup.Go(func() error {
+		if input.AdditionalInfo != nil {
+			action.Feedback = input.AdditionalInfo
+		}
+
+		_, errCreatingAction := store.CreateAction(ctx, &action)
+		if errCreatingAction != nil {
+			return errCreatingAction
+		}
+
+		return nil
+	})
+
+	updatedTime := time.Now()
+	intake.UpdatedAt = &updatedTime
+
+	// update workflow state
+	intake.Step = models.SystemIntakeStepDECISION
+	intake.State = models.SystemIntakeStateCLOSED
+	intake.DecisionState = models.SIDSLcidIssued
+
+	// update LCID-related fields when they are set
+	if input.ExpiresAt != nil {
+		intake.LifecycleExpiresAt = input.ExpiresAt
+	}
+	if input.Scope != nil {
+		intake.LifecycleScope = input.Scope
+	}
+	if input.NextSteps != nil {
+		intake.DecisionNextSteps = input.NextSteps
+	}
+	if input.CostBaseline != nil {
+		intake.LifecycleCostBaseline = null.StringFromPtr(input.CostBaseline)
+	}
+
+	var updatedIntake *models.SystemIntake // declare this outside the function we pass to errGroup.Go() so we can return it
+
+	// save intake
+	errGroup.Go(func() error {
+		var errUpdateIntake error // declare this separately because if we use := on next line, compiler thinks we're declaring a new updatedIntake variable as well
+		updatedIntake, errUpdateIntake = store.UpdateSystemIntake(ctx, intake)
+		if errUpdateIntake != nil {
+			return errUpdateIntake
+		}
+
+		return nil
+	})
+	// TODO: EASI-3109 will send an email from this mutation
+	// save admin note
+	if input.AdminNote != nil {
+		errGroup.Go(func() error {
+			adminNote := &models.SystemIntakeNote{
+				SystemIntakeID: input.SystemIntakeID,
+				AuthorEUAID:    adminEUAID,
+				AuthorName:     null.StringFrom(adminUserInfo.CommonName),
+				Content:        input.AdminNote,
+			}
+
+			_, errCreateNote := store.CreateSystemIntakeNote(ctx, adminNote)
+			if errCreateNote != nil {
+				return errCreateNote
+			}
+
+			return nil
+		})
+	}
+	if err := errGroup.Wait(); err != nil {
+		return nil, err
+	}
+
+	return updatedIntake, nil
+}
+
+// ConfirmLCID is used to confirm the choices of an already issued LCID. All fields are required, and should come back pre-populated by the front end with the previous answer
+func ConfirmLCID(ctx context.Context,
+	store *storage.Store,
+	fetchUserInfo func(context.Context, string) (*models.UserInfo, error),
+	input model.SystemIntakeConfirmLCIDInput,
+) (*models.SystemIntake, error) {
+
+	adminEUAID := appcontext.Principal(ctx).ID()
+
+	adminUserInfo, err := fetchUserInfo(ctx, adminEUAID)
+	if err != nil {
+		return nil, err
+	}
+
+	intake, err := store.FetchSystemIntakeByID(ctx, input.SystemIntakeID)
+	if err != nil || intake == nil {
+		return nil, err
+	}
+	err = lcidactions.IsLCIDValidToUpdate(intake)
```

Comment:
Should this call `IsLCIDValidToConfirm()`?

Reply:
Thank you, it should!